crates/sui-source-validation-service/src/lib.rs

```diff
@@ -30,16 +30,28 @@ use sui_source_validation::{BytecodeSourceVerifier, SourceMode};
 
 #[derive(Deserialize, Debug)]
 pub struct Config {
-    pub packages: Vec<Packages>,
+    pub packages: Vec<PackageSources>,
+}
+
+#[derive(Deserialize, Debug)]
+#[serde(tag = "source", content = "values")]
+pub enum PackageSources {
+    Repository(RepositorySource),
+    Directory(DirectorySource),
 }
 
 #[derive(Clone, Deserialize, Debug)]
-pub struct Packages {
+pub struct RepositorySource {
     pub repository: String,
     pub branch: String,
     pub paths: Vec<String>,
 }
 
+#[derive(Clone, Deserialize, Debug)]
+pub struct DirectorySource {
+    pub paths: Vec<String>,
+}
```

Comment:
Main change: create a variant `PackageSources` and allow config to detect the kind of source with a `source` field.